### PR TITLE
add OpenSSL 3.3.0

### PR DESCRIPTION
**Pull request by Kyle-Neale**

### CODE REVIEW

**Comment by chouquette** [comment]
Repening as part or https://datadoghq.atlassian.net/browse/BARX-389

**Comment by chouquette** [comment]
Waiting for https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/38678816 to complete before merging

File: config/software/openssl3.rb [hunk 1]
```diff
@@ -89,8 +90,6 @@
 
   command configure_command, env: env, in_msys_bash: true
 
-  patch source: "openssl-3.0.9-do-not-build-docs.patch", env: env
```
Comment (f4usto) [inline config/software/openssl3.rb:92]: What is the reasoning for this removal?

**Review by f4usto — COMMENTED** [review]
lgtm, since it is approved, I will just add an question.

Since we will be looking into the openSSL with the FIPS provider. Will these changes be possible to accommodate a more custom openssl build as proposed in [here](https://github.com/openssl/openssl/blob/openssl-3.3.1/README-FIPS.md#installing-the-fips-provider-and-using-it-with-the-latest-release)

File: config/software/openssl3.rb (hunk 1 above)
Comment (chouquette) [reply config/software/openssl3.rb:92]: I haven't checked but I suppose the patch doesn't apply anymore. I will double check

Comment (chouquette) [reply config/software/openssl3.rb]: I can't see an obvious reason not to apply the patch, I restored it and will check the pipeline

**Comment by chouquette** [comment]
> Since we will be looking into the openSSL with the FIPS provider. Will these changes be possible to accommodate a more custom openssl build as proposed in [here](https://github.com/openssl/openssl/blob/openssl-3.3.1/README-FIPS.md#installing-the-fips-provider-and-using-it-with-the-latest-release)

It will not be an issue, we can still select a different version that the default and add new configure arguments as required.

Although the simplest approach might be to add a new software definition that would stick to the FIPS validated branch, build as usual, and copy the 2 files that are needed for FIPS

File: config/software/openssl3.rb (hunk 1 above)
Comment (chouquette) [reply config/software/openssl3.rb]: I assumed openssl's Makefile was versionned but it isn't :facepalm: I opened a new [PR](https://github.com/DataDog/omnibus-software/pull/532) to have the same behavior regardless of the openssl version and will rebase this PR on top of it

**Pull request merged**